src/lib.rs

```diff
@@ -303,11 +310,24 @@ impl RegisterArrayInfo {
     }
 }
 
+impl RegisterClusterInfo {
+    fn parse(tree: &Element) -> RegisterClusterInfo {
+        RegisterClusterInfo {
+            registers: Some(tree.children.iter()
+                        .filter(|c| c.name == "register")
```

Comment:
I don't think this is correct. `<cluster>` trees can be nested. So both "register" and "cluster" should be allowed here.